sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/GroupByValueEndpointComponent.ts

```diff
@@ -49,43 +55,51 @@ export class GroupByValueEndpointComponent implements ExecutionContext {
     }
 
     const aggregateHeaders = getInitialHeader();
+    try {
```

Comment:
We are adding a try catch here (any many other places) just to set `err.body.fetchedSoFarResults = undefined;`. Can we instead modify the constructor of  `RUCapPerOperationExceededError` so that fetchedSoFarResults is alway initialized. This would reduce the change surface. Also what happens if the property is not set. it should be `undefined`. Are we dropping partial results here?

constructor(
    message: string = "Request Unit limit per Operation call exceeded",
    fetchedSoFarResults?: any[] // Set it to [] or directly `undefined`
  ) {

Reply:
@v1k1 we are implementing the exclusion of partial results for specific aggregate queries from the error response. So, we would return partial results for non-aggregate queries but aggregate ones we won't be returning anything.